projects/gloo/pkg/translator/clusters.go

```diff
@@ -156,6 +158,10 @@ func (t *translatorInstance) initializeCluster(
 		DnsRefreshRate:            dnsRefreshRate,
 		PreconnectPolicy:          preconnect,
 	}
+	// for kube gateway, use new stats name format
+	if envutils.IsEnvTruthy(constants.GlooGatewayEnableK8sGwControllerEnv) {
```

Comment:
I think we do things like this elsewhere but after looking at it I feel like we should be checking these once at the start of a translation rather than checking mid translation where hypothetically some clusters may have this set true and some set false. 

Not a problem with this pr more of a thought for our general strategies

Reply:
yeah we have this check and/or the proxy type check sprinkled throughout the codebase and it's a bit messy..but luckily we should be able to rip them out soon!

this is the only place that i saw that created a Cluster (besides some plugins that create Clusters for other purposes but with well-defined names that don't need this alt stat name)